core/controllers/feedback.py

```diff
@@ -22,6 +22,30 @@
 from core.controllers import base
 from core.domain import feedback_services
 from core.domain import suggestion_services
+from core.domain import user_services
+
+
+def get_updated_thread_dict(thread_dicts, key_id, key_username):
+    """Updates the thread dictionary by the given keys.
+
+    Args:
+        thread_dicts: list(dict). The dictionary that will be updated.
+        key_id: str. The id of the author to be fetched.
+        key_username: str. The username of the author.
+
+    Returns:
+        list(dict). The thread dictionary with usernames inserted.
+    """
+    ids = [t[key_id] for t in thread_dicts]
+    usernames = []
+
+    for i in ids:
+        usernames.append(user_services.get_username(i) if i else None)# type: ignore[no-untyped-call]
+
+    for index, t in enumerate(thread_dicts):
+        t[key_username] = usernames[index]
```

Comment:
Please make sure to remove the user ID from the dict too.

Reply:
done.